src/plugin.js

```diff
@@ -222,6 +222,35 @@ const onPlayerReady = (player) => {
   });
 };
 
+/**
+ * Sets up MediaKeys on demand
+ *
+ * @function initializeMediaKeys
+ * @param    {Object} [player]
+ *           A player object.
+ * @param    {Object} [emeOptions={}]
+ *           An object of eme plugin options.
+ */
+const initializeMediaKeys = (player, emeOptions = {}) => {
+  // TODO: this should be refactored
+  // fake an encrypted event for handleEncryptedEvent
+  const mockEncryptedEvent = {
+    initDataType: 'cenc',
+    initData: null,
+    target: player.tech_.el_
+  };
+
+  setupSessions(player);
+
+  // TODO: this should be refactored and renamed to be less tied
+  // to encrypted events
+  if (player.tech_.el_.setMediaKeys) {
+    return handleEncryptedEvent(mockEncryptedEvent, emeOptions, player.eme.sessions, player.tech_);
```

Comment:
Is it worth catching the promise rejection here since the mock is always expected behavior?

Reply:
It is expected to fail if you don't provide enough capability information for EME so I believe the error should bubble up